### PR TITLE
Add print layout for minutes

### LINKED ISSUE #52: Print layout
Minutes on myHPI are often printed. It is important to have a properly styled print version of minute pages.

Design and implement a print version of myHPI minutes. Use the [current print layout of myHPI minutes](https://myhpi.de/protokolle/2022-04-26) as reference.

Basic support:

- Proper layout with title and content; no menu, table of contents etc.
- Metadata of minutes
- Labels (and their color?)
- Links
- Images

Keep in mind/Should be supported (in the future?):

- Proper print mode for other page types of myHPI (like information pages)
- Abbreveations
- Attachements

Notice:

- Layout should use few ink and barely any colors, if at all
- Multiple organisations use minutes. Do not brand minutes solely to the FSR

### PR DESCRIPTION
**Pull request by frcroth**
Fixes #52

See some screenshots here.

Minutes in standard view:
image

Minutes for printing:
image

(Information pages were not the focus of this issue)
Information page in standard view:
image
image

### CODE REVIEW

File: myhpi/static/css/myHPI.css [hunk 1]
```diff
@@ -203,4 +203,39 @@ nav .nav-link[aria-expanded=true] {
 h1.toc-title::after {
     margin: 0;
     height: 0px;
-}
\ No newline at end of file
+}
+
+.print-shown {
+    display: none;
+}
```
Comment (SilvanVerhoeven) [inline myhpi/static/css/myHPI.css:210]: Is this used somewhere?

File: myhpi/static/css/myHPI.css [hunk 2]
```diff
@@ -203,4 +203,39 @@ nav .nav-link[aria-expanded=true] {
...
+
+@media print {
+    .print-none {
+        display: none !important;
+    }
+
+    .print-shown {
+        display: block !important;
+    }
```
Comment (SilvanVerhoeven) [inline myhpi/static/css/myHPI.css:219]: Boostrap already provides these utilities for us :)

File: myhpi/static/css/myHPI.css [hunk 3]
```diff
@@ -203,4 +203,39 @@ nav .nav-link[aria-expanded=true] {
...
+
+    nav, .toc, .toc-container {
+        display: none;
+    }
+
+    .minutes-container {
+        display: flex;
+        flex-direction: column;
+    }
```
Comment (SilvanVerhoeven) [inline myhpi/static/css/myHPI.css:228]: _Could_ be done with the bootstrap utilities as well, but giving the container some semantic is fine to me, too

File: myhpi/static/css/myHPI.css [hunk 4]
```diff
@@ -203,4 +203,39 @@ nav .nav-link[aria-expanded=true] {
...
+        display: none !important;
+    }
+
+    .print-shown {
+        display: block !important;
+    }
+
+    nav, .toc, .toc-container {
+        display: none;
+    }
```
Comment (SilvanVerhoeven) [inline myhpi/static/css/myHPI.css:223]: Bootstrap, probably

File: myhpi/core/templates/core/minutes.html [hunk 5]
```diff
@@ -2,12 +2,15 @@
...
+        <h1>{{  page.title }}</h1>
+        </div>
+        <div class="col-9 minutes-text">
             {{ parsed_md.0 }}
         </div>
-        <div class="col-3">
+        <div class="col-3 minutes-meta">
             <h4>Date</h4>
             <p>{{ page.date }}</p>
             <h4>Moderator</h4>
```
Comment (SilvanVerhoeven) [inline myhpi/core/templates/core/minutes.html:16]: What do you think about recreating the column layout of the old minutes? They use up less space and look cleaner, imo
grafik

**Review by SilvanVerhoeven — CHANGES_REQUESTED** [review]

**Comment by SilvanVerhoeven** [comment]
Could you use Arial as font? Or does @jeriox have an opinion about the font to use?

**Comment by SilvanVerhoeven** [comment]
About links: Might be nice to add the full links as footnotes

**Comment by SilvanVerhoeven** [comment]
Would you create new Issues for the additional stuff in #52 like attachements, so they don't get forgotten?

File: myhpi/static/css/myHPI.css (hunk 1 above)
Comment (frcroth) [reply myhpi/static/css/myHPI.css:210]: Used for the minutes title

File: myhpi/core/templates/core/minutes.html (hunk 5 above)
Comment (SilvanVerhoeven) [reply myhpi/core/templates/core/minutes.html:16]: Yeah, haven't checked it out yet either. Would be fine to me to open another issue for it

**Comment by Paula-Kli** [comment]
Do we already have Abbreviations? Since in printing those used to be written as the complete words in 1327 which was pretty cool - otherwise I would create an issue for that (Abbreviations in general + printing)

**Comment by frcroth** [comment]
> Do we already have Abbreviations? Since in printing those used to be written as the complete words in 1327 which was pretty cool - otherwise I would create an issue for that (Abbreviations in general + printing)

Abbreviations are implemented. Printing could also be a separate issue.

File: myhpi/static/css/myHPI.css (hunk 2 above)
Comment (frcroth) [reply myhpi/static/css/myHPI.css:219]: Yes, but they do not work for some reason for me.

Comment (frcroth) [reply myhpi/static/css/myHPI.css:219]: Ah they don't work because bootstrap is only loaded for media=screen, which can of course be changed

**Comment by frcroth** [comment]
@SilvanVerhoeven Font is now there, bootstrap classes are used.

File: myhpi/static/css/myHPI.css [hunk 6]
```diff
@@ -205,19 +205,11 @@ h1.toc-title::after {
     height: 0px;
 }
 
-.print-shown {
+.d-print-block {
     display: none;
 }
```
Comment (SilvanVerhoeven) [inline myhpi/static/css/myHPI.css:210]: Out if curiosity: Why do we need this CSS here? Shouldn't `d-none d-print-block` ([see example here](https://getbootstrap.com/docs/5.2/utilities/display/#display-in-print))

File: myhpi/static/css/myHPI.css [hunk 7]
```diff
@@ -205,19 +205,11 @@ h1.toc-title::after {
...
-        display: none !important;
-    }
-
-    .print-shown {
-        display: block !important;
-    }
-
     nav, .toc, .toc-container {
         display: none;
     }
```
Comment (SilvanVerhoeven) [inline myhpi/static/css/myHPI.css:215]: We could use `d-print-none` here

Comment (SilvanVerhoeven) [reply myhpi/static/css/myHPI.css:215]: Edit: At least for nav you are already using it

**Review by SilvanVerhoeven — CHANGES_REQUESTED** [review]

File: myhpi/static/css/myHPI.css (hunk 6 above)
Comment (frcroth) [reply myhpi/static/css/myHPI.css]: Because I didn't know

File: myhpi/static/css/myHPI.css (hunk 7 above)
Comment (frcroth) [reply myhpi/static/css/myHPI.css]: Problem with toc is that it is generated, there is no .toc element in the template that I could assign the class too. Of course you could add the class in the markdown parser that generates it, but at that point I like the css solution better.

Comment (SilvanVerhoeven) [reply myhpi/static/css/myHPI.css]: Ahh, yeah, agreed!

**Review by SilvanVerhoeven — APPROVED** [review]
Lgm! Supposed you create all the issues we talked about

**Comment by jeriox** [comment]
@SilvanVerhoeven could you approve the PR? I think you are more qualified for this :D

**Comment by jeriox** [comment]
oh. I read that as "commented on the PR" 🙈 maybe I should go to bed ^^

**Comment by frcroth** [comment]
> oh. I read that as "commented on the PR" 🙈 maybe I should go to bed ^^

Can't merge though without an approval
image

But does not have to be today obviously

**Comment by jeriox** [comment]
> > oh. I read that as "commented on the PR" 🙈 maybe I should go to bed ^^
> 
> Can't merge though without an approval image
> 
> But does not have to be today obviously

should work now

**Pull request merged**